lib/ansible/modules/crypto/openssl_privatekey.py

```diff
@@ -136,9 +149,25 @@ def __init__(self, module):
         self.path = module.params['path']
         self.mode = module.params['mode']
         self.changed = True
+        self.privatekey = None
+        self.fingerprint = {}
         self.check_mode = module.check_mode
 
 
+    def get_fingerprint(self):
+        """Generate the fingerprint of the public key. """
+
+        fingerprint = {}
+        privatekey = crypto.load_privatekey(crypto.FILETYPE_PEM, open(self.path, 'r').read())
+        publickey = crypto.dump_publickey(crypto.FILETYPE_ASN1, privatekey)
+        for algo in hashlib.algorithms:
+            f = getattr(hashlib, algo)
+            pubkey_digest = f(publickey).hexdigest()
+            fingerprint[algo] = ':'.join(s.encode('hex') for s in pubkey_digest.decode('hex'))
```

Comment:
- You are producing a hex-encoded digest, decoding it back into binary, then encoding it as hex again.
- `encode('hex')` (or rather the `hex` codec) is pretty nasty to use, as codecs as supposed to work with Unicode strings.  (I suspect that's why you end up doing the encode/decode/encode dance)

What about `fingerprint[algo] = ':'.join(pubkey_digest[i:i+2] for i in range(0, len(pubkey_digest), 2))` ?